libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -1987,6 +1986,14 @@ void WaveTrack::PasteOne(
         for (const auto& clip : track.mClips)
             if (clip->GetPlayStartTime() > t0 - (1.0 / rate))
                 clip->ShiftBy(insertDuration);
+    } else if (split)
+    {
+       const auto clipAtT0 = track.GetClipAtTime(t0);
+       const auto t = clipAtT0 ? clipAtT0->GetPlayEndTime() : t0;
+       if (!track.IsEmpty(t, t + insertDuration))
+          throw notEnoughSpaceException;
+       track.SplitAt(t0);
+       clipAtT0->ShiftBy(insertDuration);
```

Comment:
Should you reassign `clipAtT0` after `SplitAt`?  To be sure you shift the correct one and make no assumptions about how `SplitAt` mutates the clips previously present?

Reply:
Yes, that's a potential crash indeed. Will publish a fix ...